File: domain/src/main/java/com/ahn/domain/currency/usecase/SaveExchangeToCurrencyUseCase.kt
package com.ahn.domain.currency.usecase

import com.ahn.domain.currency.repository.CurrencySelectionRepository
import javax.inject.Inject

class SaveExchangeToCurrencyUseCase @Inject constructor(
    private val repository: CurrencySelectionRepository,
) {
    suspend operator fun invoke(code: String) {
        repository.saveExchangeToCurrencyCode(code)
    }
}

File: ios/Packages/Domain/Sources/Domain/Currency/UseCase/SaveExchangeToCurrencyUseCase.swift
public struct SaveExchangeToCurrencyUseCase: Sendable {
    private let repository: any CurrencySelectionRepository

    public init(repository: any CurrencySelectionRepository) {
        self.repository = repository
    }

    public func callAsFunction(_ code: String) async throws {
        try await repository.saveExchangeToCurrencyCode(code)
    }
}
